### PR TITLE
Validate date when create new event

### PR DESCRIPTION
**Pull request by dkryvi**

### CODE REVIEW

File: app/validation/events.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const Joi = require('joi');
+
+const createEventSchema = {
```
Comment (akosenko) [inline app/validation/events.js]: please use simplified name `eventSchema`

File: app/routes/events.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 const router = require('express').Router();
 const bodyParser = require('body-parser');
+const validate = require('express-validation');
 
 const { searchEvents, createEvent, deleteEvent } = require('../controllers/events');
+const { createEventSchema } = require('../validation/events');
```
Comment (akosenko) [inline app/routes/events.js]: `eventSchema` is more convieint in my own opinion

File: app/middleware/error_handler.js [hunk 3]
```diff
@@ -1,21 +1,37 @@
...
 
 const notFound = () => {
   throw new httpErrors.NotFound('Resource not found');
 };
 
+const prepareValidationError = err => ({
+  message: get(err, 'statusText', 'Expect valid parameters'),
+  errors: get(err, 'errors', []).map(error => ({
+    location: get(error, 'location'),
+    field: get(error, 'field[0]'),
```
Comment (akosenko) [inline app/middleware/error_handler.js:24]: why field[0]? I think here you should use `get(error, 'field')`

File: app/middleware/error_handler.js [hunk 4]
```diff
@@ -1,21 +1,37 @@
...
+    message: get(error, 'messages[0]'),
```
Comment (akosenko) [inline app/middleware/error_handler.js]: may be better to send all messages, not only first?

**Review by akosenko — CHANGES_REQUESTED** [review]

File: app/middleware/error_handler.js (hunk 4 above)
Comment (dkryvi) [reply app/middleware/error_handler.js]: I agree. [Fixed](https://github.com/inst-event-emitter/eventemitter-backend/pull/14/commits/29c56039bf17c26383f823e84cbe5d35d3a8eb8f#diff-6c681ebf59605d05df0cd7ad31902b30R25)

File: app/validation/events.js (hunk 1 above)
Comment (dkryvi) [reply app/validation/events.js]: I [renamed](https://github.com/inst-event-emitter/eventemitter-backend/pull/14/commits/29c56039bf17c26383f823e84cbe5d35d3a8eb8f#diff-21fcb63ee98c4e9d3ba6a929c055da48R3) schema name

File: app/routes/events.js (hunk 2 above)
Comment (dkryvi) [reply app/routes/events.js]: [Done](https://github.com/inst-event-emitter/eventemitter-backend/pull/14/commits/29c56039bf17c26383f823e84cbe5d35d3a8eb8f#diff-21fcb63ee98c4e9d3ba6a929c055da48R3)

File: app/middleware/error_handler.js (hunk 3 above)
Comment (dkryvi) [reply app/middleware/error_handler.js:24]: If I pick `field` value response will be as on picture(`field` will be an array)
image
That's why I pick only first item.
It was changed in [this](https://github.com/hapijs/joi/commit/8565afc5816ace1071dd5b1fa5a72355c6ec7df2#diff-f2d4d889f420f9ef17c4501ffd84e17f) commit in Joi library

Comment (akosenko) [reply app/middleware/error_handler.js:24]: gotcha

**Review by akosenko — APPROVED** [review]

**Pull request merged**